snuba/datasets/events_processor_base.py

```diff
@@ -157,43 +154,26 @@ def process_message(
         * processed_message: dict representing the processed column -> value(s)
         Returns `None` if the event is too old to be written.
         """
-        action_type = None
+        version = message[0]
+        if version != 2:
+            raise InvalidMessageVersion(f"Unsupported message version: {version}")
 
-        if isinstance(message, dict):
-            # deprecated unwrapped event message == insert
-            action_type = ProcessorAction.INSERT
+        # version 2: (2, type, data, [state])
```

Comment:
Going to add some tighter type constraints to this later.

Reply:
That would be really helpful